src/Filesystem.php

```diff
@@ -0,0 +1,128 @@
+    {
+        $location = $this->aliases->get($location);
+        parent::write($location, $contents, $config);
+    }
+
+    public function writeStream(string $location, $contents, array $config = []): void
+    {
+        $location = $this->aliases->get($location);
+        parent::writeStream($location, $contents, $config);
+    }
+
+    public function read(string $location): string
+    {
+        $location = $this->aliases->get($location);
+        return parent::read($location);
+    }
+
+    public function readStream(string $location)
+    {
+        $location = $this->aliases->get($location);
+        return parent::readStream($location);
+    }
+
+    public function delete(string $location): void
+    {
+        $location = $this->aliases->get($location);
+        parent::delete($location);
+    }
+
+    public function createDirectory(string $location, array $config = []): void
+    {
+        $location = $this->aliases->get($location);
+        parent::createDirectory($location, $config);
+    }
+
+    public function deleteDirectory(string $location): void
+    {
+        $location = $this->aliases->get($location);
+        parent::deleteDirectory($location);
+    }
+
+    public function copy(string $source, string $destination, array $config = []): void
+    {
+        $source = $this->aliases->get($source);
+        $destination = $this->aliases->get($destination);
+        parent::copy($source, $destination, $config);
+    }
+
+    public function move(string $source, string $destination, array $config = []): void
+    {
+        $source = $this->aliases->get($source);
+        $destination = $this->aliases->get($destination);
+        parent::move($source, $destination, $config);
+    }
+
+    public function setVisibility(string $path, string $visibility): void
+    {
+        $path = $this->aliases->get($path);
+        parent::setVisibility($path, $visibility);
+    }
+
+    public function visibility(string $path): string
+    {
+        $path = $this->aliases->get($path);
+        return parent::visibility($path);
+    }
+
+    public function mimeType(string $path): string
+    {
+        $path = $this->aliases->get($path);
+        return parent::mimeType($path);
+    }
+
+    public function lastModified(string $path): int
+    {
+        $path = $this->aliases->get($path);
+        return parent::lastModified($path);
+    }
+
+    public function listContents(string $location, bool $deep = LeagueFilesystem::LIST_SHALLOW): DirectoryListing
+    {
+        $location = $this->aliases->get($location);
+        return parent::listContents($location, $deep);
+    }
+
+    public function fileSize(string $path): int
+    {
+        $path = $this->aliases->get($path);
+        return parent::fileSize($path);
+    }
+
+    private function prepareAliases(array $aliases): void
+    {
+        if ($aliases !== []) {
+            $aliases = array_merge(['@root' => ''], $aliases);
```

Comment:
Looks useless. Isn't it?

Reply:
Not

Reply:
Why not? We no need to merge a new array of one element to empty this element...

Reply:
> Not, see my answer in the previous PR

Please repeat here :)

Reply:
https://github.com/yiisoft/files/pull/12#discussion_r391475122

Reply:
http://sandbox.onlinephpfunctions.com/code/946d72fe1cedd5bfdf4eb42182ec192b215437f1
@yiiliveext